### PR TITLE
Fixes bugs/exploits related to SMES. Reorganizes parts of the SMES file

### LINKED ISSUE #11848: Infinite cable coil using an SMES
**Problem Description**:
You can deconstruct an SMES' terminal multiple times, giving loads of cable coil from nowhere.

**What did you expect to happen**: 
You can deconstruct a terminal only once.

**What happened instead**:
Multiple deconstructions of a terminal allowed.

**Why is this bad/What are the consequences:**
Shouldn't be able to get infinite resources from anywhere.

**Steps to reproduce the problem**:
1. Open up an SMES
2. Spam-click the SMES using wirecutters
3. Watch cable coils appear from nowhere.

**When did the problem start happening**:
Probably since forever

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
N/A

### PR DESCRIPTION
**Pull request by SteelSlayer**
**What does this PR do:**
Fixes #11848

**Changelog:**
:cl:
fix: You can no longer spam dismantle a SMES terminal and get 10 cable for every progress bar
fix: The game will properly remove 10 cable from your stack after building a SMES terminal, instead of removing 0
add: A cable adding sound gets played when you add wires to create a terminal for a SMES
/:cl:

### CODE REVIEW

File: code/modules/power/smes.dm [hunk 1]
```diff
@@ -158,9 +158,44 @@
...
+		if(istype(tempLoc, /turf/space))
+			to_chat(user, "<span class='warning'>You can't build a terminal on space.</span>")
+			return 
+		else if(istype(tempLoc))
+			if(tempLoc.intact)
+				to_chat(user, "<span class='warning'>You must remove the floor plating first.</span>")
+				return 
+		
+		to_chat(user, "<span class='notice'>You start adding cable to the [src].</span>")
+		playsound(src.loc, C.usesound, 50, 1)
```
Comment (variableundefined) [inline code/modules/power/smes.dm]: implicit src

File: code/modules/power/smes.dm [hunk 2]
```diff
@@ -158,9 +158,44 @@
...
+
+		if(do_after(user, 50, target = src))
+			if(!terminal && panel_open)
+				T = get_turf(user)
+				var/obj/structure/cable/N = T.get_cable_node() //get the connecting node cable, if there's one
+				if(prob(50) && electrocute_mob(usr, N, N, 1, TRUE)) //animate the electrocution if uncautious and unlucky
+					do_sparks(5, 1, src)
+					return
+
+				C.use(10) // make sure the cable gets used up
```
Comment (variableundefined) [inline code/modules/power/smes.dm]: Might be just me being paranoid, but is there a check at any point to ensure that the cable indeed have 10 cable or more?

Comment (Shadow-Quill) [reply code/modules/power/smes.dm]: lines 157-159
```
if(C.amount < 10)
			to_chat(user, "<span class='alert'>You need more wires.</span>")
			return
```

Comment (SteelSlayer) [reply code/modules/power/smes.dm]: Yup, what Shadow pasted are the lines for that

**Pull request merged**